WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/wizards/projectregistration/ProjectCreatedEndingPage.java

```diff
@@ -58,11 +79,69 @@ protected void makeRegistration() {
 		}
 
 		successfulRegistration = true;
-		((ProjectRegistrationWizard) getWizard()).projectId = id;
+
+		((RegistrationWizard) getWizard()).setProjectId(id);
+
 		messageTitle = "New project registered!";
-		messageBody = "Your new project id "
-				+ id
-				+ " is registered.\nYou can change it and other WatchDog settings in the Eclipse preferences."
-				+ ProjectIdEnteredEndingPage.ENCOURAGING_END_MESSAGE;
+		messageBody = "Your new project id is registered: ";
+	}
+
+	@Override
+	public void setVisible(boolean visible) {
+		super.setVisible(visible);
+		if (visible) {
+			if (dynamicComposite != null) {
+				dynamicComposite.dispose();
+			}
+			makeRegistration();
+			createPageContent();
+			dynamicComposite.layout(true);
+			topComposite.layout(true);
+		}
+	}
+
+	private void createPageContent() {
+		setTitle(windowTitle);
+		if (isThisProjectWizzard()) {
```

Comment:
Can't you make the page wizard-agnostic by providing subclasses that are added to the correct wizards?